File: example-plugin/src/main/java/dev/aerogel/example/ExamplePlugin.java
package dev.aerogel.example;

import dev.aerogel.api.AerogelPlugin;
import dev.aerogel.api.PluginContext;
import dev.aerogel.api.event.server.ServerStartedEvent;
import net.minecraft.commands.Commands;
import net.minecraft.network.chat.Component;

public final class ExamplePlugin implements AerogelPlugin {
    @Override
    public void onLoad(PluginContext context) {
        context.logger().info("Example plugin loaded; data directory = " + context.dataDirectory());
        context.events().listen(ServerStartedEvent.class, event ->
            context.logger().info("The server has started."));
        context.commands().register(Commands.literal("aerogel-example")
            .executes(command -> {
                command.getSource().sendSuccess(
                    () -> Component.literal("Hello from Aerogel."), false);
                return 1;
            }));
    }

    @Override
    public void onUnload(PluginContext context) {
        context.logger().info("Example plugin unloaded.");
    }
}

File: example-plugin/src/main/kotlin/dev/aerogel/example/ExamplePlugin.kt
package dev.aerogel.example

import dev.aerogel.api.AerogelPlugin
import dev.aerogel.api.PluginContext
import dev.aerogel.api.event.EventHandler
import dev.aerogel.api.event.player.PlayerJoinEvent
import net.minecraft.network.chat.Component

class ExamplePlugin : AerogelPlugin {
    override fun onLoad(context: PluginContext) {
        // Log a message when the plugin is loaded
        context.logger().info("ExamplePlugin loaded!")

        // Event listener approach 1:
        // Register a PlayerJoinEvent listener through the plugin context
        context.events().listen(PlayerJoinEvent::class.java) { event ->
            event.player().sendSystemMessage(
                Component.literal("Hello! 1")
            )
        }
    }

    // Event listener approach 2:
    // Handle PlayerJoinEvent using the @EventHandler annotation
    @EventHandler
    fun onPlayerJoin(event: PlayerJoinEvent) {
        event.player().sendSystemMessage(
            Component.literal("Hello! 2")
        )
    }
}